### PR TITLE
[add feature] support decimal for position annotation.

### LINKED ISSUE #2493: [request feature] suggesting features for animal imaging data
Hi Nilearn team,

I’m a preclinical fMRI researcher and want to conduct the Nilearn for my data analysis as well as visualization. One major concern is most of the visualization features are too optimized for human research in terms of generating the publishable level of figures while it also provides some flexibility to plot animal data.

So hope to suggest the following features as API to develop some animal-research-friendly features. This has been posted originally at Neurostars (https://neurostars.org/t/request-feature-suggesting-features-for-animal-imaging-data/6959?u=dvm-shlee) but copying here since this seems the better place.

Low-level API guide or instruction to tweak glassy template: Since I’m an animal-oriented researcher, I’m focussing a few species that others not formally do. The code seems only allowed MNI space, and there is no option currently to add additional space for this (hard codded to use only ‘brain_schematics_XX.json’ in ‘glass_brain_files’). So I’m suggesting to tweak the original code as below.
“glass_brain.py”

line 87, add
def _get_json_and_transform(direction, dirname = None):

line 104-105, add
____if dirname is None:
____[indent] dirname = os.path.dirname(os.path.abspath(__file__))
____[indent] dirname = os.path.join(dirname, ‘glass_brain_files’)

line 167, insert
________if ‘dirname’ in kwargs.keys():
____________dirname = kwargs[‘dirname’]
____________del kwargs[‘dirname’]
________else:
____________dirname = None

line 175, add
____json_filename, transform = _get_json_and_transform(direction, dirname)

The next feature will be decimal position annotation for the smaller brain., Below is my suggestion on updating a code.
“display.py”

line 260, add
____def draw_position(self, size, bg_color, decimal=False, **kwargs):
________if decimal:
____________text = ‘%s=%.{}f’.format(decimal)
________else:
____________text = '%s=%i’
line 262, modify ‘%s=%i’ to text as below
________ax.text(0, 0, text % (self.direction, self.coord),

line 1001, insert
________# use decimal for annotate position
________if ‘decimal’ in kwargs.keys():
____________decimal = kwargs[‘decimal’]
____________del kwargs[‘decimal’]
________else:
____________decimal = False

line 1011, add
________________display_axis.draw_position(size=size, bg_color=bg_color, decimal=decimal, **kwargs)

“img_plotting.py”

line 135, add
____if ‘decimal’ in kwargs.keys():
________decimal = kwargs[‘decimal’]
________del kwargs[‘decimal’]
____else:
________decimal = False

line 194, add
________display.annotate(decimal=decimal)

After adding the above code, with passing ‘decimal=2’ on any plotting method, the image will be as below.

decimal

I haven’t tested the glass_brain, though, so I need the additional test to validate it. But I hope these features could be accepted into the Nilearn for animal researchers, and if this looks acceptable, I will generate a pull request according to your development guideline.

Thanks!

### PR DESCRIPTION
**Pull request by dvm-shlee**
Closes #2493 .

Changes proposed in this pull request:

-with this PR, nilearn plotting module can handle decimal for position annotation as an option,
simply by adding [decimal = int(X)], where int(X) indicates X decimal (e.g. decimal=2 will be 0.XX)
-

### CODE REVIEW

File: nilearn/plotting/displays.py [hunk 1]
```diff
@@ -257,9 +257,13 @@ def transform_to_2d(self, data, affine):
                              self.direction)
         return cut
 
-    def draw_position(self, size, bg_color, **kwargs):
+    def draw_position(self, size, bg_color, decimal=False, **kwargs):
```
Comment (bthirion) [inline nilearn/plotting/displays.py:260]: Since this function is in the public API, we should take this opportunity to write docstrings...

**Comment by dvm-shlee** [comment]
@bthirion 

- There needs to be a test
: the attached image below is the test results, 
TestResult

- add an entry in whats new
: could you give me a hint where I can find it? 

I'm afraid of the feature remaining invisiable since it is hidden as a kwarg. It should thus be documented. ideally, there should be an example using it.
: I've changes 'decimal' to 'decimals' to match with the option on nimpy.round(), 
and instead to use kwarg, add the argument key on the function. As you suggested, I documented it on plot_img and annotate methods.

Thank you!

**Comment by bthirion** [comment]
>     * : the attached image below is the test results,

Hi, many thx ! The test has to be included in the test suite of the library, here in
https://github.com/nilearn/nilearn/blob/master/nilearn/plotting/tests/test_displays.py

**Comment by dvm-shlee** [comment]
@bthirion  The test is included in the test suite, hope this can cover the requirement.
Thanks!

**Comment by jeromedockes** [comment]
could you merge master to fix conflicts? also, could you fix the error in the whatsnew (missing reference) ? thank you!

**Comment by dvm-shlee** [comment]
@jeromedockes I've merged the master to fix conflicts, but not sure about the missing reference. I will check after the CI is done.

**Comment by dvm-shlee** [comment]
Finally, success!

**Comment by dvm-shlee** [comment]
@jeromedockes Could you confirm that my final commit is acceptable for the push request?
Thank you!

**Comment by bthirion** [comment]
I'll merge tomorrow unless I hear otherwise.

File: doc/whats_new.rst [hunk 2]
```diff
@@ -20,20 +20,21 @@ NEW
...
+  to enable annotation of a slice coordinate position with the float.
 
 Fixes
 -----
 
 - :class:`nilearn.input_data.NiftiLabelsMasker` no longer ignores its `mask_img`
-- :func:`nilearn.masking.compute_brain_mask` has replaced
-  `nilearn.masking.compute_gray_matter_mask`. Features remained the same but
+- :func:`nilearn.masking.compute_brain_mask` has been replaced by
+  `:func: nilearn.masking.compute_gray_matter_mask`. Features remained the same but
```
Comment (jeromedockes) [inline doc/whats_new.rst:31]: why has this changed? both content and formatting of the new version are incorrect

File: doc/whats_new.rst [hunk 3]
```diff
@@ -20,20 +20,21 @@ NEW
...
   some corrections regarding its description were made in the docstring.
 
 
 Changes
 -------
 
-- `nilearn.datasets.fetch_surf_fsaverage5` has been removed.
```
Comment (jeromedockes) [inline doc/whats_new.rst:36]: why is this removed? seems unrelated to this PR

File: doc/whats_new.rst [hunk 4]
```diff
@@ -1580,7 +1581,7 @@ New features
    - The new module :mod:`nilearn.connectome` now has class
      :class:`nilearn.connectome.ConnectivityMeasure` can be useful for
      computing functional connectivity matrices.
-   - The function nilearn.connectome.sym_to_vec in same module
+   - The function :func:`nilearn.connectome.sym_to_vec` in same module
```
Comment (jeromedockes) [inline doc/whats_new.rst:1584]: unrelated change?

File: nilearn/plotting/displays.py [hunk 5]
```diff
@@ -257,9 +257,15 @@ def transform_to_2d(self, data, affine):
                              self.direction)
         return cut
 
-    def draw_position(self, size, bg_color, **kwargs):
+    def draw_position(self, size, bg_color, decimals=False, **kwargs):
+        if decimals:
+            text = '%s=%.{}f'.format(decimals)
+            coord = float(self.coord)
```
Comment (jeromedockes) [inline nilearn/plotting/displays.py:263]: why is this necessary?

File: nilearn/plotting/displays.py [hunk 6]
```diff
@@ -957,7 +963,7 @@ def add_markers(self, marker_coords, marker_color='r', marker_size=30,
 
     def annotate(self, left_right=True, positions=True, scalebar=False,
                  size=12, scale_size=5.0, scale_units='cm', scale_loc=4,
-                 **kwargs):
+                 decimals=False, **kwargs):
```
Comment (jeromedockes) [inline nilearn/plotting/displays.py:966]: why not 0 rather than False

File: nilearn/plotting/displays.py [hunk 7]
```diff
@@ -257,9 +257,15 @@ def transform_to_2d(self, data, affine):
                              self.direction)
         return cut
 
-    def draw_position(self, size, bg_color, **kwargs):
+    def draw_position(self, size, bg_color, decimals=False, **kwargs):
+        if decimals:
+            text = '%s=%.{}f'.format(decimals)
+            coord = float(self.coord)
+        else:
+            text = '%s=%i'
```
Comment (jeromedockes) [inline nilearn/plotting/displays.py:265]: not sure you need a special case for 0 decimals, `"%.0f" % coord` should give the expected result

File: nilearn/plotting/tests/test_displays.py [hunk 8]
```diff
@@ -137,6 +137,12 @@ def test_annotations():
                          scale_size=2.5,
                          scale_units='cm',
                          scale_loc=3)
+
+
+def test_position_annotation_with_decimals():
+    # Test of decimals position annotation with precision of 2
+    orthoslicer = OrthoSlicer(cut_coords=(0, 0, 0))
+    orthoslicer.annotate(positions=True, decimals=2)
```
Comment (jeromedockes) [inline nilearn/plotting/tests/test_displays.py:145]: this test doesn't seem to check the number of decimals is actually respected?

File: nilearn/plotting/img_plotting.py [hunk 9]
```diff
@@ -117,6 +117,7 @@ def _plot_img_with_bg(img, bg_img=None, cut_coords=None,
                       display_factory=get_slicer,
                       cbar_vmin=None, cbar_vmax=None,
                       brain_color=(0.5, 0.5, 0.5),
+                      decimals=False,
```
Comment (jeromedockes) [inline nilearn/plotting/img_plotting.py:120]: should `decimals` be exposed and documented in other functions e.g. plot_stat_map, plot_anat etc?

File: doc/whats_new.rst (hunk 2 above)
Comment (dvm-shlee) [reply doc/whats_new.rst:31]: @jeromedockes This is not the change I made, its happened when I merged my fork with original

File: doc/whats_new.rst (hunk 3 above)
Comment (dvm-shlee) [reply doc/whats_new.rst:36]: @jeromedockes This is not the change I made, its happened when I merged my fork with original

File: doc/whats_new.rst (hunk 4 above)
Comment (dvm-shlee) [reply doc/whats_new.rst:1584]: @jeromedockes This is not the change I made, its happened when I merged my fork with original

File: nilearn/plotting/displays.py (hunk 5 above)
Comment (dvm-shlee) [reply nilearn/plotting/displays.py:263]: I just wanted to keep original integer annotation and add a feature. but now it seems better to simplify. I will update the PR.

File: nilearn/plotting/img_plotting.py (hunk 9 above)
Comment (dvm-shlee) [reply nilearn/plotting/img_plotting.py:120]: WIll update on the next PR. Thank you for your comment!

File: nilearn/plotting/tests/test_displays.py (hunk 8 above)
Comment (dvm-shlee) [reply nilearn/plotting/tests/test_displays.py:145]: Will make the new test to check number of decimals.

File: nilearn/plotting/displays.py (hunk 7 above)
Comment (dvm-shlee) [reply nilearn/plotting/displays.py:265]: Will take your suggestion.

File: nilearn/plotting/displays.py (hunk 6 above)
Comment (dvm-shlee) [reply nilearn/plotting/displays.py:966]: 0 should be better as you pointed out. I will update it as above.

File: doc/whats_new.rst (hunk 2 above)
Comment (jeromedockes) [reply doc/whats_new.rst:31]: ah indeed it seems to have been fixed in master later, but then I wonder why it appears in the diff with the merge commit. could you merge upstream/master again to see if this disappears? thanks

**Comment by jeromedockes** [comment]
@dvm-shlee could you fix the small conflict in whats new? and then we can merge

**Comment by GaelVaroquaux** [comment]
I fixed the merge conflict. The tests are running. If they pass, I will merge.

**Comment by jeromedockes** [comment]
@dvm-shlee you mentionned some changes you were planning to make, are
you still working on this PR?

File: doc/whats_new.rst [hunk 10]
```diff
@@ -33,22 +33,23 @@ NEW
...
 
 
 Fixes
 -----
 
 - :class:`nilearn.input_data.NiftiLabelsMasker` no longer ignores its `mask_img`
-- :func:`nilearn.masking.compute_brain_mask` has replaced
-  `nilearn.masking.compute_gray_matter_mask`. Features remained the same but
+- :func:`nilearn.masking.compute_brain_mask` has been replaced by
+  `:func: nilearn.masking.compute_gray_matter_mask`. Features remained the same but
```
Comment (GaelVaroquaux) [inline doc/whats_new.rst:46]:
```suggestion
- :func:`nilearn.masking.compute_brain_mask` has replaced
  nilearn.masking.compute_gray_matter_mask. Features remained the same but
```

File: doc/whats_new.rst [hunk 11]
```diff
@@ -1604,7 +1605,7 @@ New features
    - The new module :mod:`nilearn.connectome` now has class
      :class:`nilearn.connectome.ConnectivityMeasure` can be useful for
      computing functional connectivity matrices.
-   - The function nilearn.connectome.sym_to_vec in same module
+   - The function :func:`nilearn.connectome.sym_to_vec` in same module
```
Comment (GaelVaroquaux) [inline doc/whats_new.rst:1608]:
```suggestion
   - The function nilearn.connectome.sym_to_vec in same module
```

File: nilearn/plotting/displays.py (hunk 6 above)
Comment (GaelVaroquaux) [inline nilearn/plotting/displays.py:966]:
```suggestion
                 decimals=0, **kwargs):
```

**Comment by GaelVaroquaux** [comment]
All green. Merging!!!
